src/nbl/video/CVKLogicalDevice.h

```diff
@@ -124,51 +327,472 @@ class CVKLogicalDevice final : public ILogicalDevice
+        VkBufferCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // Each pNext member of any structure (including this one) in the pNext chain must be either NULL or a pointer to a valid instance of VkBufferDeviceAddressCreateInfoEXT, VkBufferOpaqueCaptureAddressCreateInfo, VkDedicatedAllocationBufferCreateInfoNV, VkExternalMemoryBufferCreateInfo, VkVideoProfileKHR, or VkVideoProfilesKHR
+        vk_createInfo.flags = static_cast<VkBufferCreateFlags>(0); // Nabla doesn't support any of these flags
+        vk_createInfo.size = initialMreqs.vulkanReqs.size;
+        vk_createInfo.usage = initialMreqs.vulkanBufferUsageFlags;
+        vk_createInfo.sharingMode = static_cast<VkSharingMode>(initialMreqs.sharingMode); 
+
+        if (initialMreqs.queueFamilyIndices)
+        {
+            vk_createInfo.queueFamilyIndexCount = initialMreqs.queueFamilyIndices->size();
+            vk_createInfo.pQueueFamilyIndices = initialMreqs.queueFamilyIndices->data();
+        }
+
+        VkBuffer vk_buffer;
+        if (vkCreateBuffer(m_vkdev, &vk_createInfo, nullptr, &vk_buffer) == VK_SUCCESS)
+        {
+            IDriverMemoryBacked::SDriverMemoryRequirements bufferMemoryReqs = {};
+
+            // These are only temporary
+            bufferMemoryReqs.vulkanBufferUsageFlags = initialMreqs.vulkanBufferUsageFlags;
+            bufferMemoryReqs.sharingMode = initialMreqs.sharingMode;
+            bufferMemoryReqs.queueFamilyIndices = initialMreqs.queueFamilyIndices;
+
+            // Not sure if I'd actually need these
+            bufferMemoryReqs.memoryHeapLocation = initialMreqs.memoryHeapLocation;
+            bufferMemoryReqs.mappingCapability = initialMreqs.mappingCapability;
+            bufferMemoryReqs.prefersDedicatedAllocation = initialMreqs.prefersDedicatedAllocation;
+            bufferMemoryReqs.requiresDedicatedAllocation = initialMreqs.requiresDedicatedAllocation;
+
+            vkGetBufferMemoryRequirements(m_vkdev, vk_buffer, &bufferMemoryReqs.vulkanReqs);
+
+            return core::make_smart_refctd_ptr<CVulkanBuffer>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this), bufferMemoryReqs, vk_buffer);
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+
+    core::smart_refctd_ptr<IGPUBuffer> createGPUBufferOnDedMem(const IDriverMemoryBacked::SDriverMemoryRequirements& initialMreqs, const bool canModifySubData = false) override
+    {
+        core::smart_refctd_ptr<IGPUBuffer> gpuBuffer = createGPUBuffer(initialMreqs, false);
+
+        if (!gpuBuffer)
+            return nullptr;
+
+        // Todo(achal): Probably do not call getMemoryReqs at all but
+        // vkGetBufferMemoryRequirements or equivalent
+        core::smart_refctd_ptr<video::IDriverMemoryAllocation> bufferMemory =
+            allocateDeviceLocalMemory(gpuBuffer->getMemoryReqs());
+
+        if (!bufferMemory)
+            return nullptr;
```

Comment:
ok so here, what you'd do is factor out the common memory choosing (given memory reqs) part of the 5 `allocate**Memory` functions into a single protected method, and use that

Reply:
Obsolete!